Mathlib/RingTheory/MvPolynomial/NewtonIdentities.lean

```diff
@@ -0,0 +1,265 @@
+  intro a
+  rw [← filter_or, mem_filter]
+  refine' ⟨fun ha ↦ by tauto, fun ha ↦ _⟩
+  have hacard := le_iff_lt_or_eq.mp ha.2.1
+  tauto
+
+private theorem esymm_summand_to_weight (k : ℕ) (A : Finset σ) (h : A ∈ powersetLen k univ) :
+    ∑ j in A, weight σ R k (A, j) = k * (-1) ^ k * (∏ i in A, X i : MvPolynomial σ R) := by
+  simp [weight, mem_powerset_len_univ_iff.mp h, mul_assoc]
+
+private theorem esymm_to_weight (k : ℕ) : k * esymm σ R k =
+    (-1) ^ k * ∑ t in filter (fun t ↦ card t.fst = k) (pairs σ k), weight σ R k t := by
+  rw [esymm, sum_filter_pairs_eq_sum_powersetLen_sum σ R k (fun t ↦ weight σ R k t),
+    sum_congr rfl (esymm_summand_to_weight σ R k), mul_comm (k : MvPolynomial σ R) ((-1) ^ k),
+    ← mul_sum, ← mul_assoc, ← mul_assoc, ← pow_add, Even.neg_one_pow ⟨k, rfl⟩, one_mul]
+
+private theorem esymm_mul_psum_summand_to_weight (k : ℕ) (a : ℕ × ℕ) (ha : a ∈ antidiagonal k) :
+    ∑ A in powersetLen a.fst univ, ∑ j, weight σ R k (A, j) =
+    (-1) ^ a.fst * esymm σ R a.fst * psum σ R a.snd := by
+  simp only [esymm, psum_def, weight, ← mul_assoc, mul_sum]
+  rw [sum_comm]
+  refine' sum_congr rfl fun x _ ↦ _
+  rw [sum_mul]
+  refine' sum_congr rfl fun s hs ↦ _
+  rw [mem_powerset_len_univ_iff.mp hs, ← mem_antidiagonal.mp ha, add_sub_self_left]
+
+private theorem esymm_mul_psum_to_weight (k : ℕ) :
+    ∑ a in (antidiagonal k).filter (fun a ↦ a.fst < k),
+    (-1) ^ a.fst * esymm σ R a.fst * psum σ R a.snd =
+    ∑ t in filter (fun t ↦ card t.fst < k) (pairs σ k), weight σ R k t := by
+  rw [← sum_congr rfl (fun a ha ↦ esymm_mul_psum_summand_to_weight σ R k a (mem_filter.mp ha).left),
+    sum_filter_pairs_eq_sum_filter_antidiagonal_powersetLen_sum σ R k]
+
+/-- **Newton's identities** give a recurrence relation for the kth elementary symmetric polynomial
+in terms of lower degree elementary symmetric polynomials and power sums. -/
+theorem mul_esymm_eq_sum (k : ℕ) : k * esymm σ R k =
+    (-1) ^ (k + 1) * ∑ a in (antidiagonal k).filter (fun a ↦ a.fst < k),
+    (-1) ^ a.fst * esymm σ R a.fst * psum σ R a.snd := by
+  rw [esymm_to_weight σ R k, esymm_mul_psum_to_weight σ R k]
+  apply symm
+  apply sub_eq_zero.mp
+  rw [sub_eq_add_neg, neg_mul_eq_neg_mul, neg_eq_neg_one_mul ((-1 : MvPolynomial σ R) ^ k)]
+  nth_rw 2 [← pow_one (-1 : MvPolynomial σ R)]
+  rw [← pow_add, add_comm 1 k, ← left_distrib,
+    ← sum_disjUnion (disjoint_filter_pairs_lt_filter_pairs_eq σ k),
+    disjUnion_filter_pairs_eq_pairs σ k, weight_sum σ R k, neg_one_pow_mul_eq_zero_iff.mpr rfl]
+
+/-- A version of Newton's identities which may be more useful in the case that we know the values of
+the elementary symmetric polynomials and would like to calculate the values of the power sums. -/
+theorem psum_eq_mul_esymm_sub_sum (k : ℕ) (h : k > 0) : psum σ R k =
+    (-1) ^ (k + 1) * k * esymm σ R k -
+    ∑ a in ((antidiagonal k).filter (fun a ↦ a.fst < k)).filter (fun a ↦ a.fst ≥ 1),
+    (-1) ^ a.fst * esymm σ R a.fst * psum σ R a.snd := by
```

Comment:
I think this is slightly better:
```suggestion
    ∑ a in (antidiagonal k).filter (fun a ↦ a.fst ∈ Set.Ioo 0 k),
      (-1) ^ a.fst * esymm σ R a.fst * psum σ R a.snd := by
```
(and it should also eliminate the `a.fst ≥ 1` below which I would otherwise suggest tweaking)

Reply:
Done!